graphics.cpp

```diff
@@ -67,3 +46,35 @@ void Graphics::AddSprite(const Sprite& sprite, const SDL_Rect& destination,
   SDL_RenderCopy(renderer_, sprite.GetSpriteTexture(), &source, &destination);
   perf->StopTimer("render_copy");
 }
+
+void Graphics::AddSprite(AnimatedSprite& sprite, Perf* perf,
+                         std::string animation) {
+  if (sprite.GetVisible()) {
+    int frame = sprite.GetFrameIndex();
+    SDL_Rect source = sprite.GetSourceRect(animation, frame);
+    SDL_Rect destination = sprite.GetDestinationRect();
+    Vector2 offset = sprite.GetOffsets(animation);
+    destination.x = (destination.x + offset.x) % (constants::kWindowWidth - 64);
```

Comment:
Is Goku always 64 pixels wide? if so, it would be smart to make this 64 another constant in the constants namespace.